ipaserver/install/ipa_acme_manage.py

```diff
@@ -100,6 +248,135 @@ def check_san_status(self):
         cert = x509.load_certificate_from_file(paths.HTTPD_CERT_FILE)
         cainstance.check_ipa_ca_san(cert)
 
+    def pruning(self):
+        def run_pki_server(command, directive, prefix, value=None):
+            """Take a set of arguments to append to pki-server"""
+            args = [
+                'pki-server', command,
+                f'{prefix}.{directive}'
+            ]
+            if value:
+                args.extend([str(value)])
+            logger.debug(args)
+            result = run(args, raiseonerr=False, capture_output=True,
+                         capture_error=True)
+            if result.returncode != 0:
+                raise RuntimeError(result.error_output)
+            return result
+
+        def ca_config_set(directive, value,
+                          prefix='jobsScheduler.job.pruning'):
+            run_pki_server('ca-config-set', directive, prefix, value)
+            # ca-config-set always succeeds, even if the option is
+            # not supported.
+            newvalue = ca_config_show(directive)
+            if str(value) != newvalue.strip():
+                raise RuntimeError('Updating %s failed' % directive)
+
+        def ca_config_show(directive):
+            result = run_pki_server('ca-config-show', directive,
+                                    prefix='jobsScheduler.job.pruning')
+            return result.output.strip()
+
+        def config_show():
+            status = ca_config_show('enabled')
+            if status.strip() == 'true':
+                print("Status: enabled")
+            else:
+                print("Status: disabled")
+            for option in (
+                'certRetentionTime', 'certRetentionUnit',
+                'certSearchSizeLimit', 'certSearchTimeLimit',
+                'requestRetentionTime', 'requestRetentionUnit',
+                'requestSearchSizeLimit', 'requestSearchTimeLimit',
+                'cron',
+            ):
+                value = ca_config_show(option)
+                if value:
+                    print("{}: {}".format(pruning_labels[option], value))
+                else:
+                    print("{}: {}".format(pruning_labels[option],
+                                          default_pruning_options[option]))
+
+        def run_pruning():
+            """Run the pruning job manually"""
+
+            with NSSDatabase() as tmpdb:
+                print("Preparing...")
+                tmpdb.create_db()
+                tmpdb.import_files((paths.RA_AGENT_PEM, paths.RA_AGENT_KEY),
+                                   import_keys=True)
+                certs = tmpdb.list_certs()
```

Comment:
I suspect that the CA also needs to be imported in the temp NSS DB. I see the following:
```
# ipa-acme-manage pruning  --run
Preparing...
Starting job...
CalledProcessError(Command ['pki', '-C', '/tmp/tmpdrtygqqn/pwdfile.txt', '-d', '/tmp/tmpdrtygqqn', '-n', 'CN=IPA RA,O=IPA.TEST', 'ca-job-start', 'pruning'] returned non-zero exit status 255: "WARNING: UNTRUSTED ISSUER encountered on 'CN=replica1.ipa.test,O=IPA.TEST' indicates a non-trusted CA cert 'CN=Certificate Authority,O=IPA.TEST'\nTrust this certificate (y/N)? PKIException: Forbidden\n")
The ipa-acme-manage command failed.
```

Reply:
That was an error on my side. I see that you added the reminder to restart the CA in the troubleshooting, thanks!